data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -146,6 +149,14 @@ public void submitAttestation(Attestation attestation) {
             });
   }
 
+  public SafeFuture<Optional<SubmitCommitteeSignaturesResult>> submitCommitteeSignatures(
+      final List<SyncCommitteeSignature> signatures) {
+    return validatorApiChannel.sendSyncCommitteeSignatures(
+        signatures.stream()
+            .flatMap(signature -> signature.asInternalCommitteeSignature(spec).stream())
+            .collect(Collectors.toList()));
```

Comment:
I don't think this works - it will silently ignore any signatures that are from before the Altair fork and then the index reported in any errors from other signatures will wind up being wrong.

Since it's a corner case, I'd be tempted to just reject the entire submission if any signatures are from a phase0 slot.

Reply:
not sure what the desired behaviour should be if out of N submissions, 1 is prior to altair... like you said it's a corner case, but I do kind of agree swallowing them isn't ideal. I can try to make it return errors for those specific cases...

Reply:
I've gone with rejecting the entire submission. it should be very rare anyway, but I did add a regression test showing the behaviour.